src/open_company/api/companies.clj

```diff
@@ -58,6 +62,37 @@
     ;; update the company
     {:updated-company (company/put-company slug (merge original-company patch-updates) user)}))
 
+;; ----- Validations -----
+
+(defn malformed-post-req?
```

Comment:
Seems this schema validation really belongs at the company resource level, not here. It's not really JSON/REST API specific. It's really checking if the company map is valid. Both doesn't belong at this level, and we want that validation in the Clojure API too.